docs/src/main/tut/typeclasses/traverse.md

```diff
@@ -5,235 +5,138 @@ section: "typeclasses"
 
-### Playing with `Reader`
-Another interesting effect we can use is `Reader`. Recall that a `Reader[E, A]` is
-a type alias for `Kleisli[Id, E, A]` which is a wrapper around `E => A`.
-
-If we fix `E` to be some sort of environment or configuration, we can use the
-`Reader` applicative in our traverse.
-
-```tut:silent
-import cats.data.Reader
-
-trait Context
-trait Topic
-trait Result
-
-type Job[A] = Reader[Context, A]
-
-def processTopic(topic: Topic): Job[Result] = ???
+  def pure[A](a: A): Id[A] = Id(a)
+}
 ```
 
-We can imagine we have a data pipeline that processes a bunch of data, each piece of data
-being categorized by a topic. Given a specific topic, we produce a `Job` that processes
-that topic. (Note that since a `Job` is just a `Reader`/`Kleisli`, one could write many small
-`Job`s and compose them together into one `Job` that is used/returned by `processTopic`.)
+Now we can implement `map` by wrapping and unwrapping `Id` as necessary.
 
-Corresponding to our bunches of data are bunches of topics, a `List[Topic]` if you will.
-Since `Reader` has an `Applicative` instance, we can `traverse` over this list with `processTopic`.
+```tut:book:silent
+import cats.Functor
 
-```tut:silent
-def processTopics(topics: List[Topic]): Job[List[Result]] = 
-  topics.traverse(processTopic)
-```
+trait Traverse[F[_]] extends Functor[F] {
+  def traverse[G[_]: Applicative, A, B](fa: F[A])(f: A => G[B]): G[F[B]]
 
-Note the nice return type - `Job[List[Result]]`. We now have one aggregate `Job` that when run,
-will go through each topic and run the topic-specific job, collecting results as it goes.
-We say "when run" because a `Job` is some function that requires a `Context` before producing
-the value we want.
-
-One example of a "context" can be found in the [Spark](http://spark.apache.org/) project. In
-Spark, information needed to run a Spark job (where the master node is, memory allocated, etc.)
-resides in a `SparkContext`. Going back to the above example, we can see how one may define
-topic-specific Spark jobs (`type Job[A] = Reader[SparkContext, A]`) and then run several
-Spark jobs on a collection of topics via `traverse`. We then get back a `Job[List[Result]]`,
-which is equivalent to `SparkContext => List[Result]`. When finally passed a `SparkContext`,
-we can run the job and get our results back.
-
-Moreover, the fact that our aggregate job is not tied to any specific `SparkContext` allows us
-to pass in a `SparkContext` pointing to a production cluster, or (using the exact same job) pass
-in a test `SparkContext` that just runs locally across threads. This makes testing our large
-job nice and easy.
-
-Finally, this encoding ensures that all the jobs for each topic run on the exact same cluster.
-At no point do we manually pass in or thread a `SparkContext` through - that is taken care for us
-by the (applicative) effect of `Reader` and therefore by `traverse`.
-
-## Sequencing
```

Comment:
Since the `Traverse` tutorial starts with `Future.sequence`, is it still worth mentioning the `sequence` method in `Traverse`?